### PR TITLE
Add basic liquid template engine

### PR DESCRIPTION
**Pull request by yufeih**
#3417 

customized liquid tags TBD

### CODE REVIEW

File: src/docfx/template/LiquidTemplate.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        private static Template LoadTemplate(string fullPath)
+        {
+            var template = Template.Parse(File.ReadAllText(fullPath));
+            template.MakeThreadSafe();
+            return template;
+        }
+
+        private static object ToHash(JToken token)
```
Comment (live1206) [inline src/docfx/template/LiquidTemplate.cs]: >ToHash [](start = 30, length = 6)

Add more tests for different scenarios? for instance, the null token

**Comment by superyyrrzz** [comment]
Is this for ideal or legacy? If for legacy, it not required as liquid rendering happens on the dynamic rendering side. DHS only need the HTML applied with Mustache template.
